test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs

```diff
@@ -0,0 +1,183 @@
+
+            result.UrlHelper = urlHelper;
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, httpContext.Response.StatusCode);
+            Assert.Same(value, result.Value);
+        }
+
+        [Fact]
+        public async Task AcceptedAtActionResult_ReturnsStatusCode_SetsLocationHeader()
+        {
+            // Arrange
+            var expectedUrl = "testAction";
+            var httpContext = GetHttpContext();
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(expectedUrl);
+
+            // Act
+            var result = new AcceptedAtActionResult(
+                actionName: expectedUrl,
+                controllerName: null,
+                routeValues: null,
+                value: null);
+
+            result.UrlHelper = urlHelper;
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, httpContext.Response.StatusCode);
+            Assert.Equal(expectedUrl, httpContext.Response.Headers["Location"]);
+        }
+
+        [Fact]
+        public async Task AcceptedAtActionResult_ThrowsOnNullUrl()
+        {
+            // Arrange
+            var httpContext = GetHttpContext();
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(returnValue: null);
+
+            var result = new AcceptedAtActionResult(
+                actionName: null,
+                controllerName: null,
+                routeValues: null,
+                value: null);
+
+            result.UrlHelper = urlHelper;
+
+            // Act & Assert
+            await ExceptionAssert.ThrowsAsync<InvalidOperationException>(
+                async () => await result.ExecuteResultAsync(actionContext),
+            "No route matches the supplied values.");
+        }
+
+        private static ActionContext GetActionContext(HttpContext httpContext)
+        {
+            var routeData = new RouteData();
+            routeData.Routers.Add(Mock.Of<IRouter>());
+
+            return new ActionContext(httpContext,
+                                    routeData,
+                                    new ActionDescriptor());
+        }
+        private static HttpContext GetHttpContext()
+        {
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.PathBase = new PathString("");
+            httpContext.Response.Body = new MemoryStream();
+            httpContext.RequestServices = CreateServices();
+            return httpContext;
+        }
+
+        private static IServiceProvider CreateServices()
+        {
+            var options = new TestOptionsManager<MvcOptions>();
+            options.Value.OutputFormatters.Add(new StringOutputFormatter());
+            options.Value.OutputFormatters.Add(new JsonOutputFormatter(
```

Comment:
My suggestion would be that we use of Mock, or write a "test formatter". The reason is that we don't really care about the data that's output, just that formatters ran. Putting JSON and things like that in our tests is unnecessary coupling. We have another pending work item to remove reference to the JSON package from `.Core.Tests` so this would have to be addressed some day anyway.

Talk to me if you need help with this suggestion.